lib/Value/Matrix.pm

```diff
@@ -749,27 +952,171 @@ sub column {
 	}
 	return if $j + 1 > $d[1] || $j < -$d[1];
 	my @col = ();
-	foreach my $row (@{ $M->data }) { push(@col, $self->make($row->data->[$j])) }
+	for my $row (@{ $M->data }) { push(@col, $self->make($row->data->[$j])) }
 	return $self->make(@col);
 }
 
-#
-#  Extract a given element from the matrix
-#
+=head3 C<element>
+
+Extract an element from the given row/col.
+
+Usage:
+
+    my $A    = Matrix([ [ 1, 2, 3, 4 ], [ 5, 6, 7, 8 ], [ 9, 10, 11, 12 ] ]);
+    $A->element(2,3); # returns 7
+
+    my $B = Matrix([ [ [ 1, 2 ], [ 3, 4 ] ], [ [ 5, 6 ], [ 7, 8 ] ] ]);
+    $B->element(1,2,1); # returns 3;
+
+    my $row = Matrix([4,3,2,1]);
+    $row->element(2); # returns 3;
+=cut
+
 sub element {
 	my $self = (ref($_[0]) ? $_[0] : shift);
 	my $M    = $self->promote(shift);
 	return $M->extract(@_);
 }
 
-# @@@ assign @@@
+=head3 C<setElement>
+
+Assign an element in the matrix to a value.
+
+Inputs: indices as an arrayref and the value in a form that can be parsed.
+
+Note: this mutates the matrix itself.
+
+Usage:
+
+    my $A = Matrix([ [ 1, 2, 3, 4 ], [ 5, 6, 7, 8 ], [ 9, 10, 11, 12 ] ]);
+    $A->setElement([2,3],-5);
+
+
+=cut
+
+sub setElement {
+	my ($self, $ind, $value) = @_;
+
```

Comment:
Maybe there should be a check that the length of the `$ind` that is passed matches the dimension of the matrix.

Reply:
I'll add this.